services/agent/tests/conftest.py

```diff
@@ -187,9 +188,8 @@ def caplog_info_debug(
 
 
 @pytest.fixture(scope="module")
-def mocked_s3_server_url(mocked_aws_server: ThreadedMotoServer) -> HttpUrl:
+def mocked_s3_server_url(mocked_aws_server: ThreadedMotoServer) -> HttpUrlLegacy:
```

Comment:
This `HttpUrlLegacy` has been introduced to be compliant with the previous `HttpUrl` (Pydantic v1). It solves the issue with the trailing slash.

See: https://docs.pydantic.dev/latest/migration/#url-and-dsn-types-in-pydanticnetworks-no-longer-inherit-from-str